crates/rpc/rpc-types/src/eth/engine/error.rs

```diff
@@ -0,0 +1,54 @@
+//! Commonly used errors for the `engine_` namespace.
+
+/// List of Engine API errors, see <https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#errors>
+#[derive(Debug, Copy, PartialEq, Eq, Clone, thiserror::Error)]
+pub enum EngineRpcError {
```

Comment:
a few of them will be redundant because they will be handled by jsonrpsee directly, like MethodNotFound or 

but it's good to have all of them in a single enum so we can eventually use these instead (need to figure out how to map jsonrpsee errors like serde errors)